lib/stages/timeDownsample.cpp

```diff
@@ -127,13 +136,21 @@ void timeDownsample::main_thread() {
 
             // Accumulate integration totals
             output_frame.fpga_seq_total += frame.fpga_seq_total;
+            output_frame.rfi_total += frame.rfi_total;
 
             // Move to next frame
             nframes += 1;
             mark_frame_empty(in_buf, unique_name.c_str(), frame_id);
             frame_id = (frame_id + 1) % in_buf->num_frames;
 
         } else {
+
+            timespec output_age = std::get<1>(output_frame.time) - std::get<1>(frame.time);
```

Comment:
I may just be confused here, but given we ended up in this block (i.e. `fpga_seq_start > wdw_end`) `frame` is from a later time than `output_frame` so this quantity will always be negative and we will never enter the `if` statement. Should there be a separate `else if` for the case where the frame is too old?

Reply:
Ooops. Good point. They should be the other way around!

Reply:
And this is why we do code reviews :)